File: 백준/Gold/1806. 부분합/부분합.py
import sys
input = sys.stdin.readline
INF = 1e9

N, S = map(int, input().split())
sequence = list(map(int, input().split()))

left, right = 0, 0
sum = 0
min_length = INF

while True:
    if sum >= S:
        min_length = min(min_length, right - left)
        sum -= sequence[left]
        left += 1

    elif right == N:
        break

    else:
        sum += sequence[right]
        right += 1

if min_length == INF:
    print(0)
else:
    print(min_length)

File: 백준/Gold/1806. 부분합/부분합.java
import java.io.*;
import java.util.*;

public class Main {
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    StringTokenizer st;

    void solution() throws Exception {
        st = new StringTokenizer(br.readLine());
        int N = Integer.parseInt(st.nextToken());
        int S = Integer.parseInt(st.nextToken());

        int[] nums = new int[N];
        st = new StringTokenizer(br.readLine());

        for (int i = 0; i < N; i++) {
            nums[i] = Integer.parseInt(st.nextToken());
        }

        int left = 0;
        int right = 0;
        int sum = 0;
        int res = Integer.MAX_VALUE;

        while (true) {
            if (sum >= S) {
                res = Math.min(res, right - left);
                sum -= nums[left];
                left++;
            } else if (right == N) {
                break;
            } else {
                sum += nums[right];
                right++;
            }
        }

        if (res == Integer.MAX_VALUE) {
            System.out.println(0);
        } else {
            System.out.println(res);
        }
    }

    public static void main(String[] args) throws Exception {
        new Main().solution();
    }
}